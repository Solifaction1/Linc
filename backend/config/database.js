const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
