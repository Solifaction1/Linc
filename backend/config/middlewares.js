module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
