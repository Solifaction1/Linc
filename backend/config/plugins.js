module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
