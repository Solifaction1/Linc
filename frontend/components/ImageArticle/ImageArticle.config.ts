import ImageArticleData from '@models/imageArticle.model'

const defaultProps: ImageArticleData = {
	color: 'blueGray',
	imagePosition: 'right',
	image: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				alternativeText: 'alternativeText',
				width: 0,
				height: 0,
				hash: 'hash',
				ext: 'ext',
				mime: 'mime',
				size: 0,
				url: 'https://via.placeholder.com/400x500',
			},
		},
	},
	title: 'Find out what our customers have to say.',
	description: {
		text: '<p>description imageACTA</p>',
		className: 'string',
	},
	cta: {
		id: 0,
		title: 'cta title',
		link: '/link',
		externalLink: false,
		buttonType: 'Button primary',
	},
}

export { defaultProps }
