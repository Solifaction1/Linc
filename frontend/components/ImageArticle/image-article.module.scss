@import 'styles/vars';
@import 'styles/utilities';

.imageArticle {
	margin-top: calc(#{$spacing-base} * 18);
	margin-bottom: calc(#{$spacing-base} * 8);
	background-color: $color-gray;

	@media (max-width: $break-lg) {
		margin-top: calc(#{$spacing-base} * 55);
	}

	@media (max-width: $break-md) {
		margin-top: calc(#{$spacing-base} * 42.5);
	}

	.imageArticleContainer {
		@extend %page-container;
		position: relative;
		z-index: 3;
		grid-template-areas: 'headertext headertext headertext headertext headertext . headerimg headerimg headerimg headerimg headerimg headerimg';

		@media (max-width: $break-lg) {
			grid-template-areas:
				'headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg'
				'headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext';
			padding-bottom: unset;
		}

		&::after {
			content: '';
			position: absolute;
			top: 0;
			right: calc(49% - 6.4rem);
			z-index: 2;

			width: 7.57rem;
			height: 7.5rem;

			background: $color-gray;

			opacity: 0.9;

			mix-blend-mode: multiply;

			@media (max-width: $break-lg) {
				top: 0;
				right: unset;
				left: $page-bleed;

				width: 4.6rem;
				height: 4.6rem;
			}

			@media (max-width: $break-s) {
				left: $page-bleed-mobile;
			}
		}

		.imageArticleImage {
			position: absolute;
			top: -14.5rem;
			grid-area: headerimg;

			width: 50vw;
			height: calc(100% + 20.9rem);

			@media (max-width: $break-lg) {
				position: relative;
				top: -10.5rem;
				display: block;

				width: 100%;
				height: 25rem;
			}

			@media (max-width: $break-md) {
				top: 17.7rem;
				height: 35rem;
			}

			img {
				position: relative;
				object-fit: cover;

				width: 100%;
				max-width: 72rem;
				height: 100%;

				object-position: left;

				@media (max-width: $break-lg) {
					width: calc(100% + calc(#{$spacing-base} * 4.5));
					max-width: 100rem;
					height: auto;
					max-height: 45rem;
					margin-top: -36.3rem;

					object-position: top;
				}

				@media (max-width: $break-md) {
					top: -20.3rem;
					height: 100%;
					max-height: 35rem;
				}

				@media (max-width: $break-s) {
					width: calc(100% + calc(#{$spacing-base} * 3));
				}
			}
		}

		.imageArticleText {
			position: relative;
			grid-area: headertext;

			@media (max-width: $break-lg) {
				margin-top: -27.7rem;
				padding: 7.5rem 0;
			}

			@media (max-width: $break-md) {
				margin-top: -37.9rem;
			}

			button {
				margin-top: calc(#{$spacing-base} * 4);
			}

			h2 {
				@extend %h2;
			}

			.text p {
				@extend %body-copy-1;
				margin-top: calc(#{$spacing-base} * 4);
			}
		}
	}

	// ---IMAGE POSITION


	&.right {
		.imageArticleImage {
			@media (max-width: $break-hd) {
				text-align: right;
			}
		}
	}

	&.left {
		@media (max-width: $break-lg) {
			margin-top: calc(#{$spacing-base} * 55);
		}

		@media (max-width: $break-md) {
			margin-top: calc(#{$spacing-base} * 42.5);
		}

		.imageArticleContainer {
			grid-template-areas: 'headerimg headerimg headerimg headerimg headerimg headerimg . headertext headertext headertext headertext headertext ';

			@media (max-width: $break-lg) {
				grid-template-areas:
					'headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg headerimg'
					'headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext headertext';
			}

			&::after {
				right: unset;
				left: calc(49% - 6.4rem);

				@media (max-width: $break-lg) {
					right: $page-bleed;
					bottom: 0;
					left: unset;

					width: 4.6rem;
					height: 4.6rem;
				}
			}

			.imageArticleText {
				grid-area: headertext;
				@media (max-width: $break-md) {
					margin-top: -37.9rem;
				}
			}

			.imageArticleImage {
				right: 0;
				left: unset;
				grid-area: headerimg;

				width: 50vw;

				@media (min-width: $break-hd) {
					text-align: right;
				}

				@media (max-width: $break-lg) {
					position: relative;
					display: block;

					width: 100%;
					height: 25rem;
				}

				@media (max-width: $break-md) {
					top: 19.7rem;
					height: 35rem;
				}

				img {
					@media (max-width: $break-lg) {
						position: absolute;
						left: calc(#{$spacing-base} * (-4.5));
					}

					@media (max-width: $break-s){
						left: calc(#{$spacing-base} * (-3));
					}
				}
			}
		}
	}

	// ---BACKGROUND COLORS

	&.beige {
		background-color: $color-beige;

		.imageArticleContainer::after {
			background: $color-beige;
		}

		.imageArticleContainer .imageArticleText h2,
		.imageArticleContainer .imageArticleText li, 
		.imageArticleContainer .imageArticleText ul, 
		.imageArticleContainer .imageArticleText ol, 
		.imageArticleContainer .imageArticleText .text p {
			color: $color-beige-dark;
		}
	}

	&.blueGray {
		background-color: $color-blue-gray;

		.imageArticleContainer::after {
			background: $color-blue-gray;
		}

		.imageArticleContainer .imageArticleText h2,
		.imageArticleContainer .imageArticleText li, 
		.imageArticleContainer .imageArticleText ul, 
		.imageArticleContainer .imageArticleText ol, 
		.imageArticleContainer .imageArticleText .text p {
			color: $color-blue-gray-dark;
		}
	}

	&.darkGreen {
		background-color: $color-green-dark;

		.imageArticleContainer::after {
			background: $color-green-dark;
		}

		.imageArticleContainer .imageArticleText h2,
		.imageArticleContainer .imageArticleText li, 
		.imageArticleContainer .imageArticleText ul, 
		.imageArticleContainer .imageArticleText ol, 
		.imageArticleContainer .imageArticleText .text p {
			color: $color-green;
		}
	}

	&.lightGreen {
		background-color: $color-green;

		.imageArticleContainer::after {
			background: $color-green;
		}

		.imageArticleContainer .imageArticleText h2,
		.imageArticleContainer .imageArticleText li, 
		.imageArticleContainer .imageArticleText ul, 
		.imageArticleContainer .imageArticleText ol, 
		.imageArticleContainer .imageArticleText .text p {
			color: $color-green-dark;
		}
	}
}
