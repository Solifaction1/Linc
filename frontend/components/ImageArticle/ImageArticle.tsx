import Image from 'next/image'
import React from 'react'
import ImageArticleData from '@models/imageArticle.model'
import SwitchCTA from '@templates/SwitchCTA'
import YGWYS from '@components/YGWYS/YGWYS'
import { getColor } from '@constants/color'
import { defaultProps } from './ImageArticle.config'
import styles from './image-article.module.scss'

const ImageArticle = (props: ImageArticleData) => {
	const image = props.image.data && props.image?.data?.attributes

	return (
		<article
			className={`${styles.imageArticle} ${
				styles[getColor(props.color) || 'blue-gray']
			} ${styles[props.imagePosition]}`}
		>
			<div className={styles.imageArticleContainer}>
				{image && (
					<div className={styles.imageArticleImage}>
						<Image
							src={image.url ?? ''}
							alt={image.name ?? ''}
							width={image.width}
							height={image.height}
							priority
							loader={() => {
								return `/api/custom-loader?url=${encodeURIComponent(
									image.url
								)}?q=${100}`
							}}
						/>
					</div>
				)}
				<div className={styles.imageArticleText}>
					<h2>{props.title}</h2>
					{props.description && (
						<YGWYS text={props.description.text} className={styles.text} />
					)}

					{props.cta && (
						<SwitchCTA ctaData={props.cta} color={getColor(props.color)} />
					)}
				</div>
			</div>
		</article>
	)
}

ImageArticle.defaultProps = defaultProps

export default ImageArticle
