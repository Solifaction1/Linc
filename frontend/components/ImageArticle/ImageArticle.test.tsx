import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import ImageArticle from './ImageArticle'
import { defaultProps } from './ImageArticle.config'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('ImageArticle', () => {
	it('renders an ImageArticle', () => {
		const { container, getByText } = render(<ImageArticle {...defaultProps} />)

		expect(container).toMatchSnapshot()

		expect(container.querySelector('article')).toBeVisible()
		expect(container.getElementsByClassName('imageArticle')[0]).toBeVisible()
		expect(container.getElementsByClassName('right')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('imageArticleContainer')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('imageArticleImage')[0]
		).toBeVisible()
		const img = container.querySelector('img')
		expect(img).toBeVisible()
		expect(img?.getAttribute('alt')).toEqual('name')
		expect(img?.getAttribute('width')).toEqual('0')
		expect(img?.getAttribute('height')).toEqual('0')
		expect(img?.getAttribute('style')).toEqual('color: transparent;')
		expect(img?.getAttribute('src')).toEqual(
			'/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100'
		)

		expect(
			container.getElementsByClassName('imageArticleText')[0]
		).toBeVisible()
		expect(
			getByText('Find out what our customers have to say.', { selector: 'h2' })
		).toBeVisible()
		expect(container.getElementsByClassName('text')[0]).toBeVisible()
		expect(container.getElementsByClassName('ygwys')[0]).toBeVisible()
		expect(getByText('description imageACTA', { selector: 'p' })).toBeVisible()
		const btn = container.querySelector('button')
		expect(
			container.getElementsByClassName('button false primary md false gray')[0]
		).toBeVisible()
		expect(btn?.getAttribute('type')).toEqual('button')
		expect(getByText('cta title', { selector: 'p' })).toBeVisible()
	})
})
