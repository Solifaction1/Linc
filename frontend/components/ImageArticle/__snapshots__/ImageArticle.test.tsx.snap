// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageArticle renders an ImageArticle 1`] = `
<div>
  <article
    class="imageArticle gray right"
  >
    <div
      class="imageArticleContainer"
    >
      <div
        class="imageArticleImage"
      >
        <img
          alt="name"
          data-nimg="1"
          decoding="async"
          height="0"
          src="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100"
          srcset="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F400x500?q=100 1x"
          style="color: transparent;"
          width="0"
        />
      </div>
      <div
        class="imageArticleText"
      >
        <h2>
          Find out what our customers have to say.
        </h2>
        <div
          class="ygwys 
			text undefined"
        >
          <p>
            description imageACTA
          </p>
        </div>
        <button
          class="button false primary md false  gray"
          type="button"
        >
          <p
            class="label"
          >
            cta title
          </p>
        </button>
      </div>
    </div>
  </article>
</div>
`;
